crates/generate_blog/src/main.rs

```diff
@@ -0,0 +1,221 @@
+        aliases,
+        extra: front_matter::Extra {
+            team,
+            team_url,
+            release,
+        },
+        ..Default::default()
+    };
+
+    let contents = format!(
+        "+++\n{front_matter}+++\n{MARKDOWN_BOILERPLATE}",
+        front_matter = toml::to_string_pretty(&front_matter).unwrap(),
+    );
+
+    let colocate = Confirm::new("Are you planning to include images in the post?")
+        .with_help_message(
+            "
+  To include images in a post, the post will be stored in <slug>/index.md,
+  instead of the usualy <slug>.md. Images can then be stored in the directory
+  <slug>/ right next to the post itself. They can be included with a relative
+  link, e.g. ![alt text](my_impage.png).
+",
+        )
+        .with_default(false)
+        .prompt()?;
+
+    let base_path = format!("content/{blog_path}{slug}");
+
+    let path_ending = if colocate { "/index.md" } else { ".md" };
+
+    let mut path = PathBuf::from(format!("{base_path}{path_ending}"));
+
+    'try_write_file: {
+        if fs::create_dir_all(path.parent().unwrap()).is_ok()
+            && !path.exists()
+            && fs::write(&path, &contents).is_ok()
+        {
+            break 'try_write_file;
+        }
+        // A blog with that slug already exists. Generate an unambiguous path.
+        for i in 2.. {
+            path = PathBuf::from(format!("{base_path}@{i}{path_ending}"));
+            if fs::create_dir_all(path.parent().unwrap()).is_ok()
+                && !path.exists()
+                && fs::write(&path, &contents).is_ok()
+            {
+                break 'try_write_file;
+            }
+        }
+    }
+
+    let path = path.display();
+    println!(
+        "
+Success! A new blog post has been generated at the following path:
+
+{path}
+
+Remember: A post's date of publication is embedded in the `path` key of the
+front matter. Keep the generated placeholder (9999/12/31) until the exact date
+of publication is known. This is to prevent a post with an incorrect date from
+being published - CI checks against the placeholder.
+"
+    );
+
+    Ok(())
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+enum Blog {
+    #[default]
+    Main,
+    InsideRust,
+}
+
+impl Blog {
+    fn path(&self) -> &'static str {
+        match self {
+            Blog::Main => "",
+            Blog::InsideRust => "inside-rust/",
+        }
+    }
+}
+
+impl Display for Blog {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Blog::Main => write!(f, "The main Rust blog"),
+            Blog::InsideRust => write!(f, r#"The "Inside Rust" blog"#),
+        }
+    }
+}
+
+fn try_parse_version_from_title(title: &str) -> Option<String> {
+    // assuming there won't ever be a Rust 2.0
+    let (_, rest) = title.split_once("1.")?;
+    let (minor, rest) = rest.split_once('.')?;
+    let patch = rest[0..1].parse::<u8>().unwrap_or_default();
```

Comment:
this should use `str::get` in case that character isn't ASCII

Reply:
Unrealistic but ok :)